iceoryx_posh/include/iceoryx_posh/experimental/popo/base_publisher.hpp

```diff
@@ -0,0 +1,125 @@
+// Copyright (c) 2020 by Robert Bosch GmbH. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef IOX_EXPERIMENTAL_POSH_POPO_BASE_PUBLISHER_HPP
+#define IOX_EXPERIMENTAL_POSH_POPO_BASE_PUBLISHER_HPP
+
+#include "iceoryx_posh/experimental/popo/publishable_sample.hpp"
+#include "iceoryx_posh/internal/popo/ports/publisher_port_user.hpp"
+
+namespace iox
+{
+namespace popo
+{
+using uid_t = UniquePortId;
+
+///
+/// @brief The PublisherInterface class defines the publisher interface used by the Sample class to make it generic.
+/// This allows any publisher specialization to be stored as a reference by the Sample class.
+///
+template <typename T>
+class PublisherInterface
+{
+  public:
+    virtual void publish(Sample<T> sample) noexcept = 0;
```

Comment:
I think this should be a rvalue like: `void publish(Sample<T>&& sample) noexcept = 0;`